DietPi-Dashboard | Enable password protection with global software password by default

nonnorm opened the pull request:
<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->
**Status**: Ready
- [x] 0.3.2 dashboard release

**Reference**: https://github.com/ravenclaw900/DietPi-Dashboard/issues/11

**Commit list/description**:
<!--
- DietPi-Config | Add "Fan control" option to "Performance Options"
![Screenshot](https://xxx.github.com/images/xxx.png)
- DietPi-Config | Add "Fan control" support for Odroid C2
- DietPi-Config | Syntax fix
-->
- DietPi-Software | Add password protection to config file

nonnorm committed:
v7.9
+ DietPi-Software | Add password protection to config file

nonnorm committed:
v7.9
+ DIetPi-Software | Put quotes around hash and secret

MichaIng committed:
v7.9

+ DietPi-Dashboard | Comment config settings which match the default and add add doc comments about the default value as well
+ DietPi-Dashboard | Align hash generation with other cases, use "sha512sum" which is guaranteed to exist compared to "shasum", quote password within the command substitution to avoid globbing and word splitting

nonnorm committed:
v7.9
+ DietPi-Software | Download config file from repository

nonnorm committed:
v7.9
+ DietPi-Software | Download nightly config file from tag name, instead of release

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -3526,31 +3526,20 @@ _EOF_
-_EOF_
+			G_CONFIG_INJECT '#pass[[:blank:]]=' 'pass = true' /opt/dietpi-dashboard/config.toml
+			G_CONFIG_INJECT '#hash[[:blank:]]=' "hash = \"$(echo -n "$GLOBAL_PW" | sha512sum | mawk '{print $1}')\"" /opt/dietpi-dashboard/config.toml
+			G_CONFIG_INJECT '#secret[[:blank:]]=' "secret = \"$(openssl rand -hex 32)\"" /opt/dietpi-dashboard/config.toml
```
We must not reset the password and the secret on a reinstall. What about defining the URL for the config based on the choice and then do the download and config changes in a separate `[[ ! -f '/opt/dietpi-dashboard/config.toml' ]]` block below?

nonnorm committed:
v7.9
+ DietPi-Software | Only change config file if it doesn't exist

nonnorm committed:
v7.9
+ DietPi-Software | Config file getting enhancements

nonnorm committed:
v7.9
+ DietPi-Software | Get correct tag name

nonnorm committed:
v7.9
+ DietPi-Software | Move download of config file

nonnorm committed:
v7.9
+ DietPi-Software | Replace underscore with hyphen

MichaIng committed:
v7.9

+ DietPi-Dashboard | Minor order, code and comments

MichaIng committed:
v7.9

+ CHANGELOG | DietPi-Dashboard: On fresh installs, password protection is now enabled by default, using the global software password. You can apply or change this manually by following the instructions in our documentation: https://dietpi.com/docs/software/system_stats/#dietpi-dashboard
